### PR TITLE
Modulize GeneratedAttributeMethods

### PR DESCRIPTION
**Pull request by ksss**
Declare a method in the module that is automatically generated from the column name in Rails and include it.
This way, there is no possibility of conflicting declarations, and we can override any declarations in the model.

It also solves the problem of https://github.com/pocke/rbs_rails/issues/190

### CODE REVIEW

File: lib/rbs_rails/active_record.rb [hunk 1]
```diff
@@ -422,9 +425,12 @@ def will_save_change_to_#{col.name}?: () -> bool
             def restore_#{col.name}!: () -> void
             def clear_#{col.name}_change: () -> void
           EOS
-          sig << "attr_accessor #{col.name}? (): #{class_name}\n" if col.type == :boolean
```
Comment (ksss) [inline lib/rbs_rails/active_record.rb:425]: `#{col.name}?` method defined on https://github.com/rails/rails/blob/1a240cf83e03527567dea68f3c6bea5301d90113/activerecord/lib/active_record/attribute_methods/query.rb

There is no condition for this method to be defined.
Also, methods such as `model.foo? = false` are not generated. Therefore, it should not be an attr_accessor.

File: lib/rbs_rails/active_record.rb [hunk 2]
```diff
@@ -405,7 +406,9 @@ def #{singular}_id: () -> Integer?
           class_name_opt = optional(class_name)
           column_type = col.null ? class_name_opt : class_name
           sig = <<~EOS
-            attr_accessor #{col.name} (): #{column_type}
+            def #{col.name}: () -> #{column_type}
+            def #{col.name}=: (#{column_type}) -> #{column_type}
```
Comment (ksss) [inline lib/rbs_rails/active_record.rb:410]: This is in line with the method definition in Rails.

File: lib/rbs_rails/active_record.rb [hunk 3]
```diff
@@ -396,7 +396,8 @@ def #{singular}_id: () -> Integer?
       end
 
       private def columns
-        klass.columns.map do |col|
+        mod_sig = +"module GeneratedAttributeMethods\n"
```
Comment (ksss) [inline lib/rbs_rails/active_record.rb:399]: https://github.com/rails/rails/blob/1a240cf83e03527567dea68f3c6bea5301d90113/activerecord/lib/active_record/attribute_methods.rb#L48

**Comment by ksss** [comment]
I reported that CI was failing on ruby head because it seemed to be a bug in CRuby.
https://bugs.ruby-lang.org/issues/18185

**Review by pocke — APPROVED** [review]
LGTM :+1: Thanks!

**Pull request merged**